### PR TITLE
feat: add encryption and salt secrets

### PR DESCRIPTION
**Pull request by BrianKopp**

### CODE REVIEW

File: modules/bigeye/main.tf [hunk 1]
```diff
@@ -2212,6 +2212,40 @@ resource "aws_secretsmanager_secret_version" "robot_password" {
   secret_string = random_password.robot_password[0].result
 }
 
+resource "random_password" "base_encryption" {
+  count   = local.create_base_dw_encryption_secret ? 1 : 0
+  length  = 16
```
Comment (szhongren) [inline modules/bigeye/main.tf:2217]: should we have this be longer?

Comment (BrianKopp) [reply modules/bigeye/main.tf:2217]: Yes, good catch thank you

File: modules/bigeye/locals.tf [hunk 2]
```diff
@@ -151,6 +155,8 @@ locals {
       MQ_BROKER_PASSWORD     = local.rabbitmq_user_password_secret_arn
       MYSQL_PASSWORD         = local.datawatch_rds_password_secret_arn
       ROBOT_PASSWORD         = local.robot_password_secret_arn
+      BASE_ENCRYPTION_STRING = local.base_datawatch_encryption_secret_arn
```
Comment (szhongren) [inline modules/bigeye/locals.tf:158]: should change to BASE_ENCRYPTION_SECRET

File: modules/bigeye/variables.tf [hunk 3]
```diff
@@ -1059,6 +1059,18 @@ variable "datawatch_robot_password_secret_arn" {
   default     = ""
 }
 
+variable "datawatch_base_encryption_secret_arn" {
+  description = "ARN for secretsmanager secret holding the base encryption secret. One will be created if not provided."
```
Comment (deeno35) [inline modules/bigeye/variables.tf:1063]: Maybe 1 more sentence in here (and the salt) for what the encryption secret and salt is used for will probably be helpful

**Review by deeno35 — APPROVED** [review]

File: modules/bigeye/variables.tf (hunk 3 above)
Comment (szhongren) [reply modules/bigeye/variables.tf:1063]: They are used together to generate the AES symmetric key to actually do the encryption and decryption with

Comment (szhongren) [reply modules/bigeye/variables.tf:1063]: Generation is done in Java code

Comment (BrianKopp) [reply modules/bigeye/variables.tf]: How's that look?

**Pull request merged**